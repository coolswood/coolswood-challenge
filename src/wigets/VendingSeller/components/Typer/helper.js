import React from "react";

export const renderNumbers = () => {
    let nbs = [1,2,3,4,5,6,7,8,9,0];

    return nbs.map(i => {
        return
    })
}