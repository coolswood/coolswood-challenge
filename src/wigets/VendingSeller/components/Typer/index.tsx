import React, {FC, MouseEvent} from "react";
import CSS from 'csstype';

const wrapper: CSS.Properties = {
    display: 'flex',
    flexWrap: "wrap",
    width: "370px",
    justifyContent: 'flex-end'
}

const styles = {
    button: {
        padding: 30,
        border: '1px solid black',
        width: 60,
        cursor: 'pointer'
    },
    buyBnt: {
        marginTop: 30,
        border: '1px solid black',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        padding: 20,
        cursor: 'pointer'
    }
};

type TTyper = {
    onClickNumber: (text: number) => void,
    onClickBuy: () => void,
    onReset: () => void,
    isCorrectNmb: null | boolean
}

const Typer: FC<TTyper> = ({onClickNumber, onClickBuy, isCorrectNmb, onReset}) => {

    const clickNumber = (e: MouseEvent<HTMLElement>) => {
        onClickNumber(+e.currentTarget.innerText)
    }

    const renderNumbers = () => {
        let nbs = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0];

        return nbs.map(i => {
            return (
                <div
                    key={i}
                    onClick={clickNumber}
                    style={styles.button}
                >{i}</div>
            )
        })
    }

    return (
        <>
            <div style={wrapper}>
                {
                    renderNumbers()
                }
                <div
                    onClick={onReset}
                    style={styles.button}
                >Очистить
                </div>
            </div>
            {isCorrectNmb && <div onClick={onClickBuy} style={styles.buyBnt}>Купить</div>}
        </>
    )
}

export default Typer;