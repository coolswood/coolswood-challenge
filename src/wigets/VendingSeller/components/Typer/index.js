import React from "react";
import {renderNumbers} from "./helper";

const styles = {
    wrapper: {
        display: 'flex',
        flexWrap: 'wrap',
        width: 370,
        justifyContent: 'flex-end'
    },
    button: {
        padding: 30,
        border: '1px solid black',
        width: 60,
        cursor: 'pointer'
    },
    buyBnt: {
        marginTop: 30,
        border: '1px solid black',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        padding: 20,
        cursor: 'pointer'
    }
};

export default ({onClickNumber, onClickBuy, isCorrectNmb, onReset}) => {

    const clickNumber = (e) => {
        onClickNumber(e.target.innerText)
    }

    const renderNumbers = () => {
        let nbs = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0];

        return nbs.map(i => {
            return (
                <div
                    key={i}
                    onClick={clickNumber}
                    style={styles.button}
                >{i}</div>
            )
        })
    }

    return (
        <>
            <div style={styles.wrapper}>
                {
                    renderNumbers()
                }
                <div
                    onClick={onReset}
                    style={styles.button}
                >Очистить
                </div>
            </div>
            {isCorrectNmb && <div onClick={onClickBuy} style={styles.buyBnt}>Купить</div>}
        </>
    )
}