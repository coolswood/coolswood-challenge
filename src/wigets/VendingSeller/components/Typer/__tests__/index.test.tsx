import React from "react";
import {render, fireEvent} from '@testing-library/react';
import Typer from "../index";

describe('Typer', () => {
    it('isCorrectNmb: null', () => {
        const {container} = render (
            <Typer
                onClickNumber={() => {}}
                onClickBuy={() => {}}
                isCorrectNmb={null}
                onReset={() => {}}
            />
        )

        expect(container).toMatchSnapshot()
    })

    it('isCorrectNmb: true', () => {
        const onClickNumber = jest.fn();
        const onClickBuy = jest.fn();
        const onReset = jest.fn();

        const {container, getByText} = render (
            <Typer
                onClickNumber={onClickNumber}
                onClickBuy={onClickBuy}
                isCorrectNmb={true}
                onReset={onReset}
            />
        )

        expect(container).toMatchSnapshot()

        fireEvent.click(getByText(/1/));

        expect(onClickNumber.mock.calls.length).toBe(1);

        fireEvent.click(getByText(/Купить/));

        expect(onClickBuy.mock.calls.length).toBe(1);

        fireEvent.click(getByText(/Очистить/));

        expect(onReset.mock.calls.length).toBe(1);

    })

    it('isCorrectNmb: false', () => {
        const {container} = render (
            <Typer
                onClickNumber={() => {}}
                onClickBuy={() => {}}
                isCorrectNmb={true}
                onReset={() => {}}
            />
        )

        expect(container).toMatchSnapshot()
    })
})