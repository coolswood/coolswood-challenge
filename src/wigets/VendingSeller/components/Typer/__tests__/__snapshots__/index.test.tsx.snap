// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typer isCorrectNmb: false 1`] = `
<div>
  <div
    style="display: flex; flex-wrap: wrap; width: 370px; justify-content: flex-end;"
  >
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      1
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      2
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      3
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      4
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      5
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      6
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      7
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      8
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      9
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      0
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      Очистить
    </div>
  </div>
  <div
    style="margin-top: 30px; border: 1px solid black; display: flex; justify-content: center; align-items: center; padding: 20px; cursor: pointer;"
  >
    Купить
  </div>
</div>
`;

exports[`Typer isCorrectNmb: null 1`] = `
<div>
  <div
    style="display: flex; flex-wrap: wrap; width: 370px; justify-content: flex-end;"
  >
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      1
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      2
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      3
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      4
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      5
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      6
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      7
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      8
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      9
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      0
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      Очистить
    </div>
  </div>
</div>
`;

exports[`Typer isCorrectNmb: true 1`] = `
<div>
  <div
    style="display: flex; flex-wrap: wrap; width: 370px; justify-content: flex-end;"
  >
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      1
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      2
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      3
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      4
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      5
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      6
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      7
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      8
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      9
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      0
    </div>
    <div
      style="padding: 30px; border: 1px solid black; width: 60px; cursor: pointer;"
    >
      Очистить
    </div>
  </div>
  <div
    style="margin-top: 30px; border: 1px solid black; display: flex; justify-content: center; align-items: center; padding: 20px; cursor: pointer;"
  >
    Купить
  </div>
</div>
`;
